src/detect-engine-address.c

```diff
@@ -1009,6 +1014,38 @@ static int DetectAddressParse2(const DetectEngineCtx *de_ctx,
     return -1;
 }
 
+/**
+ * \internal
+ * \brief Wrapper function for address parsing to minimize heap allocs during address parsing.
+ *
+ * \retval Return value from DetectAddressParseInternal
+ */
+static int DetectAddressParse2(const DetectEngineCtx *de_ctx, DetectAddressHead *gh,
+        DetectAddressHead *ghn, const char *s, int negate, ResolvedVariablesList *var_list,
+        int recur)
+{
+    int rc;
+#define MAX_ADDRESS_LENGTH 8192
+
+    size_t address_length = strlen(s);
+    if (address_length > (MAX_ADDRESS_LENGTH - 1)) {
+        char *address = SCCalloc(1, address_length);
```

Comment:
Is there a way to get rid of `address` and use only `s` ?

Reply:
`address` is a return parameter ... built as `s` is parsed.

Reply:
Ok, what is this return parameter for ?

Reply:
Recursion is possible when parsing addresses; when this happens `address` contains the subset of the original address to be parsed in the next step.

Reply:
What recursion ? Between variables like `EXTERNAL_NET` and their values ?

Reply:
See the calls in `DetectAddressParseInternal` that invoke `DetectAddressParse2` for example.

Reply:
I think I do not get it :-/
So I would upvote code simplicity over saving an allocation...

Reply:
From what I can gather (and remember), `address` is really only used to tokenize the input string on the fly. Other than making a copy the only modifications I can spot are inserting `\0` chars to split the string before passing the parts on to further parsing logic. If this is indeed the case another approach could be to pass arrays of chars with an explicit length around. It does look like this is quite intrusive as we call out into various functions with the address part strings as the argument.